### PR TITLE
Stricter Condition validation, remove legacy OR condition support

### PR DESCRIPTION
**Pull request by mvorisek**
buggy now, good impulse to accept more strict input only

### CODE REVIEW

File: src/Model/Scope/Condition.php [hunk 1]
```diff
@@ -200,8 +209,8 @@ public function toQueryArguments(): array
                         } else {
                             $refModel->addCondition($field, $operator, $value);
                             $field = $refModel->action('exists');
-                            $operator = null;
-                            $value = null;
+                            $operator = '>';
```
Comment (mvorisek) [inline src/Model/Scope/Condition.php:212]: needed for MSSQL

File: tests/ScopeTest.php [hunk 2]
```diff
@@ -173,7 +173,7 @@ public function testContitionToWords()
 
         $country = clone $this->country;
 
-        $country->addCondition('Users/#');
+        $country->addCondition('Users/#', '>', 0);
```
Comment (mvorisek) [inline tests/ScopeTest.php:176]: needed for MSSQL, see https://dba.stackexchange.com/questions/275313/can-an-expression-be-used-as-an-expression-as-well-as-a-predicate

**Comment by mvorisek** [comment]
@georgehristov please fix the toWords

**Comment by georgehristov** [comment]
Shall we drop support for legacy OR so fast?

**Review by georgehristov — APPROVED** [review]
LGTM

**Pull request merged**